src/jobs/models/jobManager.ts

```diff
@@ -79,9 +80,12 @@ export class JobManager {
     return { jobId, isResettable };
   }
 
-  public async resetJob(req: IJobsParams): Promise<void> {
+  public async resetJob(req: IResetJobRequest): Promise<void> {
     const jobId = req.jobId;
-    await this.transactionManager.resetJob(jobId);
+    const newExpirationDate = req.newExpirationDate;
+    const newExpirationDateString = newExpirationDate === undefined ? 'undefined' : newExpirationDate.toString();
```

Comment:
why not:
newExpirationDate as string
or
newExpirationDate ?? 'undefiend'

this seams overcomplicated.
those options can also be used directly in the log as it isnt used in any other place and it will make the need of this conversion more understandable